tools/generatechangelog/generatechangelog.go

```diff
@@ -0,0 +1,245 @@
+// Package main contains script that generates changes for the latest version.
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	"github.com/google/go-github/v57/github"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	OrgOwner = "FerretDB"
+	Repo     = "FerretDB"
+)
+
+// The PR itself from the Github endpoint
+type PRItem struct {
+	URL    string `json:"html_url"`
+	Number int    `json:"number"`
+	Title  string `json:"title"`
+	User   struct {
+		Login string `json:"login"`
+	} `json:"user"`
+	Labels []struct {
+		Name string `json:"name"`
+	} `json:"labels"`
+}
+
+// The deconstructed template categories from the given template file
+type ReleaseTemplate struct {
+	Changelog struct {
+		Categories []TemplateCategory `yaml:"categories"`
+	} `yaml:"changelog"`
+}
+
+type TemplateCategory struct {
+	Title  string   `yaml:"title"`
+	Labels []string `yaml:"labels"`
+}
+
+// An intermediate struct to group PRs by labels and categories
+type GroupedPRs struct {
+	CategoryTitle string
+	PRs           []PRItem
+}
+
+type CategorizedPRs struct {
+	Groups []GroupedPRs
+}
+
+func NewGitHubClient() *github.Client {
+	return github.NewClient(nil).WithAuthToken(os.Getenv("GITHUB_TOKEN"))
+}
+
+func GetLatestTag(ctx context.Context, client *github.Client, owner, repo string) (*github.RepositoryTag, error) {
+	tags, _, err := client.Repositories.ListTags(ctx, owner, repo, &github.ListOptions{PerPage: 1})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list tags: %w", err)
+	}
+
+	if len(tags) == 0 {
+		return nil, fmt.Errorf("no tags found in the repository")
+	}
+
+	return tags[0], nil
+}
+
+func GetCommitDate(ctx context.Context, client *github.Client, owner, repo, commitSHA string) (*time.Time, error) {
+	commit, _, err := client.Git.GetCommit(ctx, owner, repo, commitSHA)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commit: %w", err)
+	}
+
+	return &commit.Author.Date.Time, nil
+}
+
+func FetchPRs(ctx context.Context, client *github.Client, owner, repo, sinceDate string) ([]PRItem, error) {
+	query := fmt.Sprintf("repo:%s/%s is:pr is:merged merged:>=%s", owner, repo, sinceDate)
+	opts := &github.SearchOptions{Sort: "created", Order: "desc", ListOptions: github.ListOptions{PerPage: 100}}
+
+	issues, _, err := client.Search.Issues(ctx, query, opts)
```

Comment:
Understood. Just to be clear, this would be the flow:

1. Get the milestone based on title
2. Search for PRs which are closed and merged based on the milestone number
3. Format according to release template

Reply:
Re 2 – not “search”. Search is a different and more complicated API. We just need to get a list of PRs within the milestone: https://github.com/FerretDB/FerretDB/milestone/63?closed=1

Reply:
Understood. I used the "Issues" API instead, which has a `ListByRepo` method which allows to pass the milestone number and filter by closed. Subsequently, I identified Issues as Pull Requests by using the `Issue::IsPullRequest` method. However, I'm not sure about the distinction between Closed and Merged Pull requests, so I will take a further look at the API to clarify this.